lib/full-site-editing/block-templates.php

```diff
@@ -105,10 +105,20 @@ function _gutenberg_build_template_result_from_file( $template_file, $template_t
  *
  * @param WP_Post $post Template post.
  *
- * @return WP_Block_Template Template.
+ * @return WP_Block_Template|WP_Error Template.
  */
 function _gutenberg_build_template_result_from_post( $post ) {
-	$theme = get_the_terms( $post, 'wp_theme' )[0]->slug;
+	$terms = get_the_terms( $post, 'wp_theme' );
+
+	if ( is_wp_error( $terms ) ) {
+		return $terms;
+	}
+
+	if ( ! $terms ) {
+		return new WP_Error( 'template_part_missing_theme', __( 'No theme is defined for this template part.', 'gutenberg' ) );
```

Comment:
This could be both a template or template part. I think using "template" is more generic.